core/src/main/java/com/klaytn/caver/contract/Contract.java

```diff
@@ -106,7 +106,11 @@ public Contract deploy(ContractDeployParams deployParam, SendOptions sendOptions
      */
     public Contract deploy(ContractDeployParams deployParam, SendOptions sendOptions, TransactionReceiptProcessor processor) throws IOException, TransactionException {
         List<Type> deployParams = deployParam.getDeployParams();
-        this.constructor.checkTypeValid(deployParams);
+
+        if(constructor != null) {
```

Comment:
When can a contract not have a constructor?

Reply:
It can make a smart contract without constructor.